libraries/AP_OSD/AP_OSD_MSP_DisplayPort.cpp

```diff
@@ -77,6 +76,12 @@ uint8_t AP_OSD_MSP_DisplayPort::format_string_for_osd(char* buff, uint8_t size,
 
 void AP_OSD_MSP_DisplayPort::flush(void)
 {
+    if (!_bInitedUart) {
```

Comment:
Can you explain this? There are some threading issues in the way the uart is set up, but I am not sure which one you are trying to fix here.

Needs formatting

Reply:
If the internal uart variable thread owner does not match the current thread read function fails.
Look at the read code. TSo uart init must get called from the thread of the task that is going to do the actual reading.
As it was written before uart init was called from the scope of a differnt thread leading to the receive not working.
he available does not have a thread check since its a const function,
but read does.  This leads to an infinite loop where availible says you have data, but read returns -1 as the read thread check fails. The fact this could get messed up in the future is why I added the -1 check to the read function in the other file.
As for format do you guys have clang formatter  template? I'm an allman  brace code guy and I find the traiilng brace really hard to read... and have no idea what needs done to fix formatting to your standard neither the brace placement nor indenting make any sense to me at all it looks like random gibberish to me... my ability to fix formatting to your standard is basically non existent,

Reply:
Ok, makes sense but needs a comment

https://ardupilot.org/dev/docs/style-guide.html#style-guide

Reply:
There is no problem with available() being const because the thread ownership function is in C (actually it might just be a C) macro. That said it is *not* const - so this argument doesn't apply. Either way they *must* be consistent so https://github.com/ArduPilot/ardupilot/pull/20963 is the right way to do this